pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/OsteoConsentVersion2.java

```diff
@@ -0,0 +1,104 @@
+
+
+public class OsteoConsentVersion2 implements CustomTask {
+
+    private static final Logger LOG = LoggerFactory.getLogger(OsteoConsentVersion2.class);
+    private static final String DATA_FILE = "patches/consent-version-2.conf";
+    private static final String OSTEO_STUDY = "CMI-OSTEO";
+
+    private Config dataCfg;
+    private Instant timestamp;
+    private String versionTag;
+    private Config cfg;
+
+    @Override
+    public void init(Path cfgPath, Config studyCfg, Config varsCfg) {
+        File file = cfgPath.getParent().resolve(DATA_FILE).toFile();
+        if (!file.exists()) {
+            throw new DDPException("Data file is missing: " + file);
+        }
+        dataCfg = ConfigFactory.parseFile(file);
+
+        if (!studyCfg.getString("study.guid").equals(OSTEO_STUDY)) {
+            throw new DDPException("This task is only for the " + OSTEO_STUDY + " study!");
+        }
+        cfg = studyCfg;
+        versionTag = dataCfg.getString("versionTag");
+        timestamp = Instant.now();
+    }
+
+    @Override
+    public void run(Handle handle) {
+        User adminUser = handle.attach(UserDao.class).findUserByGuid(cfg.getString("adminUser.guid")).get();
+
+        String activityCode = dataCfg.getString("activityCode");
+        StudyDto studyDto = handle.attach(JdbiUmbrellaStudy.class).findByStudyGuid(dataCfg.getString("study.guid"));
+
+        LOG.info("Changing version of {} to {} with timestamp={}", activityCode, versionTag, timestamp);
+        long ts = this.timestamp.toEpochMilli();
+        updateVariables(handle, adminUser, studyDto, activityCode, ts);
+    }
+
+    private void updateVariables(Handle handle, User adminUser, StudyDto studyDto, String activityCode, long ts) {
+        String VARIABLES_V2 = "translation-updates";
+        List<? extends Config> configList = dataCfg.getConfigList(VARIABLES_V2);
+        System.out.println(configList);
+        String reason = String.format(
+                "Update activity with studyGuid=%s activityCode=%s to versionTag=%s",
+                studyDto.getGuid(), activityCode, versionTag);
+
+        RevisionMetadata meta = new RevisionMetadata(ts, adminUser.getId(), reason);
+        long activityId = ActivityBuilder.findActivityId(handle, studyDto.getId(), activityCode);
+        ActivityVersionDto version2 = handle.attach(ActivityDao.class).changeVersion(activityId, "v2", meta);
+        for (Config config : configList) {
+            revisionVariableTranslation(config.getString(TRANSLATION_KEY), config.getString(TRANSLATION_NEW), handle, meta, version2);
+        }
+    }
+
+    private static final String TRANSLATION_KEY = "varName";
+    private static final String TRANSLATION_NEW = "newValue";
+
+    private void revisionVariableTranslation(String varName, String newTemplateText, Handle handle, RevisionMetadata meta, ActivityVersionDto version2) {
+        long tmplVarId = handle.attach(SqlHelper.class).findTemplateVariableIdByVariableName(varName);
```

Comment:
make sure this works on dev environment, where there will be multiple records for same variable name

Reply:
Added order by DESC for primary key which solves this specific problem.